packages/date/src/getTimeZoneString.ts

```diff
@@ -22,23 +22,30 @@ import { getSettings } from './settings';
 /**
  * Transforms a given integer into a valid UTC Offset in hours.
  *
- * @param {number} offset A UTC offset as an integer
- * @return {string} UTC offset in hours.
+ * @param offset A UTC offset as an integer
+ * @return UTC offset in hours.
  */
-function integerToUTCOffset(offset) {
-  const offsetInHours = offset > 23 ? offset / 60 : offset;
+function integerToUTCOffset(offset: number) {
   const sign = offset < 0 ? '-' : '+';
-  const absoluteOffset = Math.abs(offsetInHours);
+  const absoluteOffset = Math.abs(offset);
+  const isHours = absoluteOffset <= 12;
+  const offsetInMinutes = isHours
+    ? Math.round((absoluteOffset % 1) * 60)
+    : Math.round(absoluteOffset % 60);
+  const offsetInHours = isHours
+    ? Math.floor(absoluteOffset)
+    : Math.floor(absoluteOffset / 60);
 
-  return offsetInHours < 10
-    ? `${sign}0${absoluteOffset}`
-    : `${sign}${absoluteOffset}`;
+  const hoursAsString = String(offsetInHours).padStart(2, '0');
+  const minutesAsString = String(offsetInMinutes).padStart(2, '0');
+
+  return `${sign}${hoursAsString}:${minutesAsString}`;
```

Comment:
Completely rewrote this to work correctly with both low and high numbers, including or excluding minutes, both negative and positive offsets. See tests for coverage.

Reply:
@swissspidy Can you confirm the change here just in case? I think we originally took this logic based on how WP shows the UTC timezone difference, it's used in the calendar only (currently), I believe:
![Screenshot 2022-09-06 at 13 47 49](https://user-images.githubusercontent.com/3294597/188616617-611a26cf-c0ee-4927-8c35-5863d05ed247.png)

Reply:
Oh yeah, that looks a bit wrong. We could add an argument about whether minutes are included? Or make sure we can display it including minutes (some actual areas do have non-integer hour offsets).

Reply:
Thanks for pointing this out.

So right now I have `GMT+2` in the date picker, and this change here would turn this into `GMT+02:00`?

Including minutes seems reasonable _if there is an offset_. Otherwise `+2` is more compact.

----

Aside: WordPress shows the timezone abbreviation instead of the offset if it exists:

<img width="294" alt="Screenshot 2022-09-06 at 10 06 03" src="https://user-images.githubusercontent.com/841956/188697231-c763e75c-5fb4-499e-bc34-46dae651982e.png">

I'll create a ticket to match that.

Reply:
I've changed it in 27d60baa5a so that minutes are only displayed if non-zero. Can you both check that this looks good in the calendar?

Reply:
So now it displays what you've input in the system general settings for your WP installation. See table below

| Timezone Setting in WP | Calendar |
|-|-|
| `"Copenhagen"` (my actual timezone as reported by my computer's clock) | <img width="359" alt="Screen Shot 2022-09-06 at 21 56 16" src="https://user-images.githubusercontent.com/637548/188726973-95c23c42-b10d-42a1-83af-252498a6eca2.png"> | 
| `"New York"` | <img width="359" alt="Screen Shot 2022-09-06 at 21 52 20" src="https://user-images.githubusercontent.com/637548/188726555-fbf67d0f-b819-4fb2-8306-fc52ce89bee9.png"> |
| `"UTC+6"` | <img width="359" alt="Screen Shot 2022-09-06 at 21 59 14" src="https://user-images.githubusercontent.com/637548/188727486-87ba5530-be1e-4094-b85d-90e2dde00a61.png"> |
| `"UTC+5:45"` | <img width="359" alt="Screen Shot 2022-09-06 at 21 58 41" src="https://user-images.githubusercontent.com/637548/188727466-d668e64d-da5d-4f2a-a5a2-637cb36a307e.png"> |
| `"UTC-4"` |  <img width="359" alt="Screen Shot 2022-09-06 at 21 52 37" src="https://user-images.githubusercontent.com/637548/188726656-b135860c-1c46-4506-8de1-4eef946f46f4.png"> |
| `"UTC-3:30"`| <img width="359" alt="Screen Shot 2022-09-06 at 21 55 45" src="https://user-images.githubusercontent.com/637548/188726956-a20737c5-729b-4f3b-b370-7ae852c7b4d4.png"> |

I guess we can all agree that the last one is a bug, which I'll look at, but otherwise it looks ok, no?

Reply:
Hm, can't recreate that last one, must have selected `-04:30` because it works correctly now. Please ignore that :P

Reply:
We could perhaps ditch the leading zero again, but otherwise looks good 👍

Reply:
So, I didn't actually add that leading zero. It was always there. The old code was this: https://github.com/GoogleForCreators/web-stories-wp/blob/5de647c3456ab5cd021a252b6c4873e5996f5c2b/packages/date/src/getTimeZoneString.js#L33-L35

This was pretty broken though, as it didn't take negative offsets into consideration (`-10 < 10` so it would become. `"-010"` 🤷 ).

Your example of `"GMT+2"` has nothing to do with this function actually. If this code was used, it would have said `"UTC"`, not `"GMT"`. So you must have selected `"GMT+2"` in the WP settings somehow as the named timezone of your system, rather than an offset as a number.

Reply:
Oh, interesting! I must check my settings then.

Then let‘s ship this 👍